dev/README_AIRFLOW3_DEV.md

```diff
@@ -0,0 +1,156 @@
+2. Are there any workarounds?
+3. Scope of change. Both in code lines / number of files and components changed.
+4. Is the feature at the heart of Airflow (scheduler, dag parser) or peripheral.
+5. Identity of the requester. Is the request from/supported by a member of the community?
+6. Similar previous cases approved.
+7. Other considerations that may raise by PMC members depending on the case.
+
+## Developing for Airflow 3
+
+PRs should target `main` branch.
+
+## Developing for Airflow 2.10.x
+
+PR should target `v2-10-test` branch. When cutting a new release for 2.10 release manager
+will sync `v2-10-test`  branch to `v2-10-stable` and cut the release from the stable branch.
+PR should never target `v2-10-stable` unless specifically instructed by release manager.
+
+## Developing for Airflow 2.11
+
+Version 2.11 is planned to be cut from `v2-10-stable` branch.
+The version will contain features relevant as bridge release for Airflow 3.
+We will not backport otherwise features from main branch to 2.11
+Note that 2.11 policy may change as 2.11 become closer
+
+# Committers / PMCs
+
+The following section explains the protocol for merging PRs.
+
+## Merging PRs for providers and Helm chart
+
+Make sure PR target `main` branch.
+Avoid merging PRs that involves providers + core / helm chart + core
+Core part should be extracted to separated PR.
+Exclusions should be pre-approved specifically with a comment by release manager.
+Do not treat PR approval (Green V) as exclusion approval.
+
+## Merging PR for Airflow 3 and 2.10.x / 2.11.x
+
+The committer who merge the PR is responsible for back-porting the PR to `v2-10-test`.
+This means start a new PR where the original commit from main is cherry-picked and conflicts resolved.
+If cherry-pick is too complex then ask the PR author / start your own PR against `v2-10-test` directly with the change.
+Note: tracking that the PRs merged as expected is the responsibility of committer who merged the PR.
+
+Committer may also request from PR author to raise 2 PRs one against `main` branch and one against `v2-10-test` prior to accepting the code change.
+
+Mistakes happen and such backport PR work might fall through cracks, so PRs that the committer thinks should be back-ported, should get 2.10.x milestone set.
+
+This way release manager can verify (as usual) if all the "expected" PRs have been backported and cherry-pick remaining PRS.
+
+## Merging PRs 2.10.x
+
+Suggested change
+Make sure PR target `v2-10-test` branch and merge it when ready.
+Make sure your PRs target the `v2-10-test` branch, and it can be merged when ready.
+All regular protocols of merging considerations are applied.
+
+## Merging PRs for Airflow 3
+
+Make sure PR target `main` branch.
+
+### PRs that involve breaking changes
+
+Make sure it has newsfragment, please allow time for community members to review.
+Our goal is to avoid breaking changes if we can so we should allow time to review, don't rush to merge.
```

Comment:
```suggestion
Our goal is to avoid breaking changes whenever possible. Therefore, we should allow time for community members to review PRs that contain such changes - please avoid rushing to merge them. In addition, ensure that these PRs include a newsfragment.
```